realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java

```diff
@@ -51,8 +51,93 @@
 import io.realm.mongodb.functions.Functions;
 
 /**
- * FIXME
+ * An <i>App</i> is the main client side entry point for interacting with a <i>MongoDB Realm App</i>.
+ *
+ * The <i>App</i> can be used to:
+ * <ul>
+ *   <li>Register uses and perform various user related operations through authentication providers
+ *   ({@link io.realm.mongodb.auth.ApiKeyAuth}, {@link EmailPasswordAuthImpl})</li>
+ *   <li>Synchronize data between the local device and a remote Realm App with Synchronized Realms</li>
+ *   <li>Invoke Realm App functions with {@link Functions}</li>
+ *   <li>Access remote data from MongoDB databases with a {@link io.realm.mongodb.mongo.MongoClient}</li>
+ * </ul>
+ * <p>
+ * To create an app that is linked with a remote <i>Realm App</i> initialize Realm and configure the
+ * <i>App</i> as below:
+ * <p>
+ * <pre>
+ *    class MyApplication extends Application {
+ *
+ *         @Override
+ *         public void onCreate() {
+ *             super.onCreate();
+ *
+ *             Realm.init(this);
+ *
+ *             AppConfiguration appConfiguration = new AppConfiguration.Builder(BuildConfig.MONGODB_REALM_APP_ID)
+ *                     .appName(BuildConfig.VERSION_NAME)
+ *                     .appVersion(Integer.toString(BuildConfig.VERSION_CODE))
+ *                     .build();
+ *
+ *             App APP = new App(appConfiguration);
+ *         }
+ *
+ *     }
+ * </pre>
+ * <p>
+ * After configuring the <i>App</i> you can start managing users, configure Synchronized Realms,
+ * call remote Realm Functions and access remote data through Mongo Collections. The below examples
+ * shows the synchronized APIs which cannot be used from the main thread. For the equivalent
+ * asynchronous counter parts. The example project in please see
```

Comment:
A couple of grammar mistakes.
```suggestion
 * call remote Realm Functions and access remote data through Mongo Collections. The examples below
 * show the synchronized APIs which cannot be used from the main thread. For the equivalent
 * asynchronous counterparts. The example project in please see
```

Also, these two sentences do not make grammatical sense:
```
For the equivalent asynchronous counterparts. The example project in please see
```